wolf/media/gst/core/w_caps.hpp

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "wolf.hpp"
+
+#include "media/gst/internal/w_common.hpp"
+#include "media/gst/internal/w_wrapper.hpp"
+#include "media/gst/core/w_structure.hpp"
+
+#include <gst/gst.h>
+
+#include <string_view>
+
+namespace wolf::gst {
+
+/**
+ * @brief wrapper of GstCaps, gstreamer capabilities concept.
+ */
+class w_caps : public w_wrapper<w_caps, GstCaps, void, gst_caps_unref>
```

Comment:
please avoid object-oriented programming style in WOLF.

Reply:
I found it inevitable unfortunately. this is a CRTP kind of inheritance, although `w_wrapper` doesn't have much to do with the derived (`Template` part in `CRTP`), but it is indeed a repeating pattern better lifted into `w_wrapper`.

It provides a common interface expressed via `internal::RawAccess`, keeping different wrappers interconnected.

I couldn't find another design without contravening DRY and SRP principles, resulting in too many and hard to keep sync/maintained repeated code.